python/lsst/summit/utils/guiders/reading.py

```diff
@@ -0,0 +1,724 @@
+        )
+
+    md = raw_stamps.metadata
+
+    # fix CCD in the metadata
+    if dayObs < 20250509:
+        md["CCDSLOT"] = ccd_swapped
+
+    # also get the ampName
+    segment = md["ROISEG"]
+    ampName = "C" + segment[7:]
+
+    # build the CCD view Bounding Boxes
+    ccd_view_bbox = mk_roi_bboxes(md, camera)
+
+    # also build the Translation methods
+    # from CCD view -> DVCS view and the reverse
+    ft, bt = mk_ccd_to_dvcs(ccd_view_bbox, detector.getOrientation().getNQuarter())
+
+    if nstamps is None:
+        nstamps = md["N_STAMPS"]
+
+    timestamps = get_timestamps(raw_stamps, nstamps)  # to populate the metadata timestamps
+    # freq = timestamps.freq * 86400.0  # seconds
+
+    goodstamps = np.where(~timestamps.mask)[0].tolist()
+    badstamps = np.where(timestamps.mask)[0].tolist()
+    stampsDict = {}
+    i = 0
+    for index in goodstamps:
+        masked_ims = raw_stamps.getMaskedImages()[i]
+        rmd = raw_stamps[i].metadata
+        rmd["DAQSTAMP"] = rmd.get("DAQSTAMP", index)
+
+        # convert to DVCS or CCD view
+        raw_roi = masked_ims.getImage().getArray()
+        roi_dvcs = convert_roi(raw_roi, md, detector, ampName, camera, view=view)
+
+        # build a Stamp Object
+        output_masked_im = MaskedImageF(roi_dvcs)
+        archive_element = [ccd_view_bbox, ft, bt]
+        stampsDict[index] = Stamp(output_masked_im, archive_element, metadata=rmd)
+        i += 1
+
+    for index in badstamps:
+        # print(
+        #     f"Warning: The stamp {i} is missing ind detector {detNum},""
+        #     "inserting empty stamp"
+        # )
+        nrows, ncols = int(md["ROIROWS"]), int(md["ROICOLS"])
+        img0 = ImageF(array=np.zeros((nrows, ncols), dtype=np.float32))
+        output_masked_im = ExposureF(MaskedImageF(img0))
+        archive_element = [ccd_view_bbox, ft, bt]
+        # create empty metadata
+        md_empty = md.toDict().copy()
+        md_empty["DAQSTAMP"] = index
+        md_empty["STMPTMJD"] = np.nan
+        stampsDict[index] = Stamp(output_masked_im, archive_element, metadata=md_empty)
+
+    stamp_list = [stampsDict[i] for i in range(nstamps)]
+    output_stamps = Stamps(stamp_list, md, use_archive=True)
+    return output_stamps
+
+
+def get_timestamps(raw_stamps: Stamps, nstamps: int = 50) -> Time:
+    timestamps_list = [stamp.metadata.get("STMPTMJD", np.nan) for stamp in raw_stamps]
+    mjd_array = np.ma.masked_invalid(timestamps_list)
+    timestamps = Time(mjd_array, format="mjd", scale="utc")
+
+    # infer frequency from valid timestamps
+    dt = np.diff(timestamps.jd)
+    freq = np.nanmedian(dt)
+    start = timestamps[0].jd
+
+    if np.isnan(freq) or freq <= 0:
+        raise ValueError(
+            f"Invalid frequency {freq} derived from timestamps. "
+            "Ensure that the timestamps are valid and evenly spaced."
+        )
+
+    timestamps_ideal = Time(start + np.arange(nstamps) * freq, format="jd", scale="utc")
+
```

Comment:
I might add a sanity check here, that once you've idealised the timestamps that the end point here and the actual last stamp are within some reasonable tolerance (not sure what, maybe 0.1s? but you know the data better than me)